cpp/tests/core/TensorFunction.cpp

```diff
@@ -234,5 +234,71 @@ TEST_P(TensorFunctionPermuteDevices, Append) {
     EXPECT_TRUE(core::Append(self, other).AllClose(self.Append(other)));
 }
 
+TEST_P(TensorFunctionPermuteDevices, Maximum) {
+    core::Device device = GetParam();
+
+    core::Tensor input, other, output;
+
+    // 0-D Tensor.
+    input = core::Tensor::Init<float>(0, device);
+    other = core::Tensor::Init<float>(1, device);
+    output = core::Maximum(input, other);
+    EXPECT_TRUE(output.AllClose(other));
+
+    // 1-D Tensor.
+    input = core::Tensor::Init<float>({2, 1, 4, 3}, device);
+    other = core::Tensor::Init<float>({4, 5, 2, 7}, device);
+    output = core::Maximum(input, other);
+    EXPECT_TRUE(
+            output.AllClose(core::Tensor::Init<float>({4, 5, 4, 7}, device)));
+
+    // 2-D Tensor.
+    input = core::Tensor::Init<float>({{2, 1}, {4, 3}}, device);
+    other = core::Tensor::Init<float>({{4, 5}, {2, 7}}, device);
+    output = core::Maximum(input, other);
+    EXPECT_TRUE(output.AllClose(
+            core::Tensor::Init<float>({{4, 5}, {4, 7}}, device)));
+
+    // Special case: Different input shape.
+    input = core::Tensor::Init<float>({2, 4, 3}, device);
+    other = core::Tensor::Init<float>({{4, 5, 2}, {3, 5, 1}}, device);
+    output = core::Maximum(input, other);
+    EXPECT_TRUE(output.AllClose(
+            core::Tensor::Init<float>({{4, 5, 3}, {3, 5, 3}}, device)));
+}
```

Comment:
at least for one case, we must also verify behavior over different dtypes such as int, uint, and bool. You may add this after the Special case.

Reply:
Should we make `bool` dtype supported? I think `LogicalAnd` and `LogicalOr` are equivalent to `Minimum` and `Maximum` for `bool` respectively. @reyanshsolis